patches/tModLoader/Terraria.UI/ItemSlot.cs.patch

```diff
@@ -17,22 +17,59 @@
  		private static bool[] canFavoriteAt;
  		private static float[] inventoryGlowHue;
  		private static int[] inventoryGlowTime;
-@@ -558,10 +_,12 @@
+@@ -556,12 +_,15 @@
+ 						}
+ 					}
  				}
- 				else if (num == 3)
- 				{
+-				else if (num == 3)
+-				{
 -					Main.mouseItem.netDefaults(inv[slot].netID);
+-					if (inv[slot].buyOnce)
+-					{
+-						Main.mouseItem.Prefix((int)inv[slot].prefix);
++				else if (PlayerHooks.CanBuyItem(player, Main.npc[player.talkNPC], inv, inv[slot])
++					&& num == 3)
++				{
 +					Main.mouseItem = inv[slot].Clone();
 +					Main.mouseItem.stack = 1;
- 					if (inv[slot].buyOnce)
- 					{
--						Main.mouseItem.Prefix((int)inv[slot].prefix);
++					if (Main.mouseItem.buyOnce)
++					{
 +						Main.mouseItem.value *= 5; // preserve item value for items sold to the shop
```

Comment:
may need to be wary here, if we can change sell values and such, *5 may not be appropriate anymore

Reply:
this is how vanilla handles buy/sellprice , by *5 / 5
modfying price should modify item.price (afaik), so this should scale appropriately

Reply:
wait, so does that mean I don't have to change anything?
and yes, the idea was that the modder should just modify item.price, which I've decided is the most optimal way.  If the modder wants to change the item itself, it's their fault if users complain, I guess.